### PR TITLE
Feature/settings + JSONL

### LINKED ISSUE #23: Feature Request: JSONL Output
Can I request jsonl output formatting? JSON is nice, but in its current form it does not ingest easily. Current JSON output example:
```json
Record 1
{
  "Event": {
    "#attributes": {
      "xmlns": "http://schemas.microsoft.com/win/2004/08/events/event"
    },
    "EventData": {
      "Data": "caller=sppsvc.exe"
    },
    "System": {
      "Channel": "Microsoft-Client-Licensing-Platform/Admin",
      "Computer": "DESKTOP-1N4R894",
      "Correlation": null,
      "EventID": 100,
      "EventRecordID": 1,
      "Execution": {
        "#attributes": {
          "ProcessID": 1348,
          "ThreadID": 1368
        }
      },
      "Keywords": "0x2000000000000001",
      "Level": 4,
      "Opcode": 0,
      "Provider": {
        "#attributes": {
          "Guid": "B6CC0D55-9ECC-49A8-B929-2B9022426F2A",
          "Name": "Microsoft-Client-Licensing-Platform"
        }
      },
      "Security": {
        "#attributes": {
          "UserID": "S-1-5-18"
        }
      },
      "Task": 0,
      "TimeCreated": {
        "#attributes": {
          "SystemTime": "2018-07-06T18:38:20.815807Z"
        }
      },
      "Version": 0
    }
  }
}
Record 2
{
  "Event": {
    "#attributes": {
      "xmlns": "http://schemas.microsoft.com/win/2004/08/events/event"
    },
    "EventData": {
      "Data": "10.0.17134.1"
    },
    "System": {
      "Channel": "Microsoft-Client-Licensing-Platform/Admin",
      "Computer": "DESKTOP-1N4R894",
      "Correlation": null,
      "EventID": 101,
      "EventRecordID": 2,
      "Execution": {
        "#attributes": {
          "ProcessID": 1348,
          "ThreadID": 1372
        }
      },
      "Keywords": "0x2000000000000001",
      "Level": 4,
      "Opcode": 0,
      "Provider": {
        "#attributes": {
          "Guid": "B6CC0D55-9ECC-49A8-B929-2B9022426F2A",
          "Name": "Microsoft-Client-Licensing-Platform"
        }
      },
      "Security": {
        "#attributes": {
          "UserID": "S-1-5-18"
        }
      },
      "Task": 0,
      "TimeCreated": {
        "#attributes": {
          "SystemTime": "2018-07-06T18:38:20.819393Z"
        }
      },
      "Version": 0
    }
  }
}
```

Proposed JSONL output:

JSONL is easier to work with because each line is the record. Example:
```json
{"Event":{"#attributes":{"xmlns":"http://schemas.microsoft.com/win/2004/08/events/event"},"EventData":{"Data":"caller=sppsvc.exe"},"System":{"Channel":"Microsoft-Client-Licensing-Platform/Admin","Computer":"DESKTOP-1N4R894","Correlation":null,"EventID":100,"EventRecordID":1,"Execution":{"#attributes":{"ProcessID":1348,"ThreadID":1368}},"Keywords":"0x2000000000000001","Level":4,"Opcode":0,"Provider":{"#attributes":{"Guid":"B6CC0D55-9ECC-49A8-B929-2B9022426F2A","Name":"Microsoft-Client-Licensing-Platform"}},"Security":{"#attributes":{"UserID":"S-1-5-18"}},"Task":0,"TimeCreated":{"#attributes":{"SystemTime":"2018-07-06T18:38:20.815807Z"}},"Version":0}}}
{"Event":{"#attributes":{"xmlns":"http://schemas.microsoft.com/win/2004/08/events/event"},"EventData":{"Data":"10.0.17134.1"},"System":{"Channel":"Microsoft-Client-Licensing-Platform/Admin","Computer":"DESKTOP-1N4R894","Correlation":null,"EventID":101,"EventRecordID":2,"Execution":{"#attributes":{"ProcessID":1348,"ThreadID":1372}},"Keywords":"0x2000000000000001","Level":4,"Opcode":0,"Provider":{"#attributes":{"Guid":"B6CC0D55-9ECC-49A8-B929-2B9022426F2A","Name":"Microsoft-Client-Licensing-Platform"}},"Security":{"#attributes":{"UserID":"S-1-5-18"}},"Task":0,"TimeCreated":{"#attributes":{"SystemTime":"2018-07-06T18:38:20.819393Z"}},"Version":0}}}
```

### PR DESCRIPTION
**Pull request by omerbenamram**
- added a `jsonl` output.
- a reference to settings is now passed to output.

closes #23

### CODE REVIEW

File: src/evtx_parser.rs [hunk 1]
```diff
@@ -75,17 +75,22 @@ pub struct EvtxParser<T: ReadSeek> {
...
 
-#[derive(Clone)]
+#[derive(Debug, Clone, PartialEq)]
 pub struct ParserSettings {
+    /// Controls the number of threads used for parsing chunks concurrently.
     num_threads: usize,
+    /// If enabled, chunk with bad checksums will be skipped.
     validate_checksums: bool,
+    /// If true, output will be indented.
+    pretty: bool,
```
Comment (ohadravid) [inline src/evtx_parser.rs]: So why not cal it 'indent'?

File: src/xml_output.rs [hunk 2]
```diff
@@ -7,45 +7,63 @@ use quick_xml::events::{BytesDecl, BytesEnd, BytesStart, BytesText, Event};
...
 
 pub trait BinXmlOutput<W: Write> {
     /// Implementors are expected to provide a `std::Write` target.
     /// The record will be written to the target.
-    fn with_writer(target: W) -> Self;
+    fn with_writer(target: W, settings: &ParserSettings) -> Self;
 
     /// Consumes the output, returning control of the inner writer to the caller.
-    fn into_writer(self) -> Result<W, Error>;
+    fn into_writer(self, settings: &ParserSettings) -> Result<W, Error>;
```
Comment (ohadravid) [inline src/xml_output.rs]: It's kind of ugly to pass the settings over and over again.
It also doesn't really make any sense to use two different settings in the same Output instance.
Maybe only pass it to `with_writer`, and let the parser copy anything it needs into it's internal state.

File: src/binxml/assemble.rs [hunk 3]
```diff
@@ -228,39 +259,48 @@ fn expand_borrowed_template<'a>(
...
     // we do not own it.
     for token in template.definition.as_ref().tokens.iter() {
-        if let BinXMLDeserializedTokens::Substitution(ref substitution_descriptor) = token
-        {
+        if let BinXMLDeserializedTokens::Substitution(ref substitution_descriptor) = token {
             if substitution_descriptor.ignore {
                 continue;
             } else {
-                let value = &template.substitution_array[substitution_descriptor.substitution_index as usize];
+                let value = &template.substitution_array
```
Comment (ohadravid) [inline src/binxml/assemble.rs]: IMO this should be considered a bug in `rustfmt`, this is just terrible 😞

Comment (omerbenamram) [reply src/binxml/assemble.rs]: sorry about that :( we could turn it off manually if it's that bad

**Pull request merged**